purchase_split_wizard/wizards/purchase_split_wizard.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, exceptions, fields, models
+
+
+class PurchaseSplitWizard(models.TransientModel):
+    _name = "purchase.split.wizard"
+    _description = "Wizard to split purchases"
+
+    purchase_origin_id = fields.Many2one(
+        string="Purchase", comodel_name="purchase.order", readonly=True
+    )
+    company_id = fields.Many2one(
+        string="Company",
+        comodel_name="res.company",
+        required=True,
+        default=lambda self: self.env.company.id,
```

Comment:
I think you should use related="purchase_origin_id.company_id". This field is used in some domains and you have to make sure that those domains take the elements from the purchase's company, not the current one, right?